compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -1709,28 +1705,37 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
         ifOverload(tree, js.Block(inner ++ tree.ctor.beforeCall ++ assignments))
       }
 
-      def postStats(tree: ConstructorTree): js.Tree = {
+      def postStats(tree: ConstructorTree[SplitSecondaryJSCtor]): js.Tree = {
         val inner = tree.subCtors.map(postStats(_))
         ifOverload(tree, js.Block(tree.ctor.afterCall ++ inner))
       }
 
+      val primaryCtor = ctorTree.ctor
+      val secondaryCtorTrees = ctorTree.subCtors
+
       js.Block(
           secondaryCtorTrees.map(preStats(_, primaryCtor.params)) ++
           primaryCtor.body ++
           secondaryCtorTrees.map(postStats(_))
       )
     }
 
+    private sealed trait JSCtor {
+      val sym: Symbol
+      val params: List[Symbol]
+    }
+
     private class PrimaryJSCtor(val sym: Symbol,
-        val params: List[Symbol], val body: List[js.Tree])
+        val params: List[Symbol], val body: List[js.Tree]) extends JSCtor
 
     private class SplitSecondaryJSCtor(val sym: Symbol,
         val params: List[Symbol], val beforeCall: List[js.Tree],
         val targetCtor: Symbol, val ctorArgs: List[js.Tree],
-        val afterCall: List[js.Tree])
+        val afterCall: List[js.Tree]) extends JSCtor
 
-    private class ConstructorTree(val lowestNum: Int, val overloadNum: Int,
-        val ctor: SplitSecondaryJSCtor, val subCtors: List[ConstructorTree])
+    private class ConstructorTree[Ctor <: JSCtor](
+        val overloadNum: Int, val ctor: Ctor,
+        val subCtors: List[ConstructorTree[SplitSecondaryJSCtor]])
```

Comment:
In order to avoid the above issue, consider defining `lo` and `hi` in `ConstructorTree`:
```scala
val lo = overloadNum
val hi = subCtors.lastOption.fold(lo)(_.hi)
```